infra/enrolment-records-store/src/test/java/com/simprints/infra/enrolment/records/store/commcare/CommCareIdentityDataSourceTest.kt

```diff
@@ -0,0 +1,404 @@
+        every { mockDataCursor.getString(0) } returnsMany
+                listOf("someOtherDatumId", "subjectActions", "someOtherDatumId", "subjectActions")
+        every { mockDataCursor.getString(1) } returnsMany
+                listOf(SUBJECT_ACTIONS_FINGERPRINT_1, SUBJECT_ACTIONS_FINGERPRINT_2)
+
+        val query = SubjectQuery()
+        val range = 0..expectedFingerprintIdentities.size
+        val actualIdentities = dataSource.loadFingerprintIdentities(query, range)
+
+        assertEquals(expectedFingerprintIdentities.size, actualIdentities.size)
+        val areContentsEqual =
+            expectedFingerprintIdentities
+                .zip(actualIdentities) { expected, actual ->
+                    expected.subjectId == actual.subjectId &&
+                            expected.fingerprints
+                                .zip(actual.fingerprints) { expectedFingerprint, actualFingerprint ->
+                                    expectedFingerprint.fingerIdentifier ==
+                                            actualFingerprint.fingerIdentifier &&
+                                            expectedFingerprint.templateQualityScore ==
+                                            actualFingerprint.templateQualityScore &&
+                                            expectedFingerprint.template.contentEquals(
+                                                actualFingerprint.template) &&
+                                            expectedFingerprint.format == actualFingerprint.format
+                                }
+                                .all { it }
+                }
+                .all { it }
+        assertTrue(areContentsEqual)
+        coVerify { mockContentResolver.query(mockMetadataUri, any(), any(), any(), any()) }
+        coVerify { mockContentResolver.query(mockDataCaseIdUri, any(), any(), any(), any()) }
+    }
+
+    @Test
+    fun testLoadFaceIdentities() = runTest {
+        every { mockMetadataCursor.count } returns expectedFaceIdentities.size
+        every { mockMetadataCursor.moveToPosition(0) } returns true
+        every { mockMetadataCursor.moveToNext() } returnsMany listOf(true, false)
+        every { mockDataCursor.moveToNext() } returns true
+        every { mockDataCursor.getColumnIndexOrThrow(COLUMN_DATUM_ID) } returns 0
+        every { mockDataCursor.getColumnIndexOrThrow(COLUMN_VALUE) } returns 1
+        every { mockDataCursor.getString(0) } returnsMany
+                listOf("someOtherDatumId", "subjectActions", "someOtherDatumId", "subjectActions")
+        every { mockDataCursor.getString(1) } returnsMany
+                listOf(SUBJECT_ACTIONS_FACE_1, SUBJECT_ACTIONS_FACE_2)
+
+        val query = SubjectQuery()
+        val range = 0..expectedFaceIdentities.size
+        val actualIdentities = dataSource.loadFaceIdentities(query, range)
+
+        assertEquals(expectedFaceIdentities.size, actualIdentities.size)
+        val areContentsEqual =
```

Comment:
Is this auto-formatting? It is really hard to read due to those new lines and spaces.

Reply:
Yeah, sorry. Will reformat.